### PR TITLE
Rs/added app state and weights component

### PR DESCRIPTION
**Pull request by SakaRicky**
- Added global app state.
- material ui theming
- created component to display fetched weights

### CODE REVIEW

File: src/services/weight.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    count: number,
+    next: number | null,
+    previous: number | null,
+    results: [BodyWeightType]
+}
+
+export const get_weights = async () => {
+    const {data: received_weights} = await axios.get<ResponseType>(BASEURL, {
+                headers: {
+                    Authorization: 'Token 31e2ea0322c07b9df583a9b6d1e794f7139e78d4'
```
Comment (rolandgeider) [inline src/services/weight.ts:16]: while this is ok for now, commiting secret keys is not a great idea :) Does react have some way of loading some kind of configuration? Then we could distinguish between development and production (where we can't hard code a specific token, but that is an issue for later)

File: src/state/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+export const setWeights = (weights: BodyWeightType[]): Action => {
+    return { type: setStateType.SET_WEIGHTS, payload: weights };
+};
+
+export const reducer = (state: State, action: Action): State => {
+    switch (action.type) {
+
+        case "SET_WEIGHTS":
```
Comment (rolandgeider) [inline src/state/reducer.ts:18]: can we use an enum here?

File: tsconfig.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "baseUrl": "./src",
     "target": "es5",
```
Comment (rolandgeider) [inline tsconfig.json:4]: do you think we coult set this to es6?

File: src/services/weight.ts (hunk 1 above)
Comment (SakaRicky) [reply src/services/weight.ts:16]: Oh my bad I forgot that.
Yeah using env variables should do it.

File: src/state/reducer.ts (hunk 2 above)
Comment (SakaRicky) [reply src/state/reducer.ts:18]: Yeah good idea.

File: tsconfig.json (hunk 3 above)
Comment (SakaRicky) [reply tsconfig.json:4]: Yeah

File: src/components/BodyWeight/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import {render} from '@testing-library/react';
+import {get_weights} from "../../services";
```
Comment (SakaRicky) [inline src/components/BodyWeight/index.test.tsx:3]: We could use absolute import here

File: src/components/BodyWeight/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import {BodyWeight} from "./index";
+
+jest.mock("axios");
+jest.mock("../../services");
```
Comment (SakaRicky) [inline src/components/BodyWeight/index.test.tsx:7]: Same here

File: src/components/BodyWeight/index.test.tsx (hunk 4 above)
Comment (rolandgeider) [reply src/components/BodyWeight/index.test.tsx:3]: Yes. I'll see if where I can configure the ide because I always just click on "add Import"

Comment (SakaRicky) [reply src/components/BodyWeight/index.test.tsx:3]: I already configured it in the tsconfig file. removing the leading ../../ should be enough.

**Pull request merged**